libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Binary/CDDL.hs

```diff
@@ -226,10 +338,93 @@ distinct x =
         |The type parameter must support .size, for example: bytes or uint
         |]
     $ "distinct_"
-      <> T.pack (show x)
+      <> show' x
         =:= (x `sized` (8 :: Word64))
         / (x `sized` (16 :: Word64))
         / (x `sized` (20 :: Word64))
         / (x `sized` (24 :: Word64))
         / (x `sized` (30 :: Word64))
         / (x `sized` (32 :: Word64))
+  where
+    show' :: Value s -> T.Text
+    show' = \case
+      VBytes -> T.pack "bytes"
+      VUInt -> T.pack "uint"
+      _ -> error "Unsupported Value for `distinct`"
+
+-- Shelley does not support some of the tagged core datastructures that we rely
+-- on in future eras. In order to have the "correct" common specification in
```

Comment:
I don't understand this comment.  Why is "correct" quoted? Isn't it actually correct for core to have these specifications? And how do you mean we override them?

Reply:
Oh I see now this definition and comment where in Shelley before, don't think it applies as is now that the definition moved to core